src/abstractdataframe/iteration.jl

```diff
@@ -440,15 +444,33 @@ julia> mapcols(x -> x.^2, df)
    2 │     4    144
    3 │     9    169
    4 │    16    196
+
+julia> mapcols(x -> x.^2, df, cols="y")
+4×2 DataFrame
+ Row │ x      y
+     │ Int64  Int64
+─────┼──────────────
+   1 │     1    121
+   2 │     2    144
+   3 │     3    169
+   4 │     4    196
 ```
 """
-function mapcols(f::Union{Function, Type}, df::AbstractDataFrame)
+function mapcols(f::Union{Function, Type}, df::AbstractDataFrame; cols=All())
+    if cols == All() || cols == Colon()
```

Comment:
How about moving this into the loop? That would avoid an allocation in the most common case, and anyway it will be optimized out if you do
```suggestion
    if cols === All() || cols === Colon()
```
But if you adopt my next suggestion then maybe you can keep the check here and just do `apply = 1:ncol(df)`.

Reply:
I thought that allocating a small vector of `true` is cheap. And this keeps the code simple. But if you feel it is important to try for maximum performance here I can refactor it to avoid this allocation.

Reply:
I guess it depends on whether it makes the code significantly more complex or not.

Reply:
I changed it to `Iterators.repeated(true)` and checked that the performance and allocations are the same in the default case as currently.

Reply:
Perfect. I hadn't though about this solution.